Desktop/analysis/analysis.cpp

```diff
@@ -386,38 +386,40 @@ Json::Value Analysis::asJSON(bool withRSource) const
 	return analysisAsJson;
 }
 
-Json::Value& Analysis::_getParentBoundValue(const QVector<JASPControl::ParentKey>& parentKeys, bool& found, bool createAnyway)
+// This method tries to find the parent keys in _boundValues Json object
+// If found, it sets the path to this reference to parentNames and returns a reference of the sub Json object
+Json::Value& Analysis::_getParentBoundValue(const QVector<JASPControl::ParentKey>& parentKeys, QVector<std::string>& parentNames, bool& found, bool createAnyway)
 {
 	found = (parentKeys.size() == 0);
-	Json::Value* result = &_boundValues;
+	Json::Value* parentBoundValue = &_boundValues;
 
-	// A parentKey has a 'name', a 'key' and a 'value': it assumes that the json boundValue is an abject, one of its member is 'name',
-	// whom value is an array of json objects. These objects have member 'key', and one of them has for value 'value': this method returns
-	// this json object (if there are several parentKeys, it repeats this operation, and returns the last json object)
+	// A parentKey has 3 properties: <name>, <key> and <value>: it assumes that the json boundValue is an abject, one of its member is <name>,
+	// whom value is an array of json objects. These objects have a member <key>, and one of them has for value <value>.
+	// if there are several parentKeys, it repeats this operation
 	//
 	// {
-	//		a : "one"
+	//		anOptionName : "one"
 	//		...
-	//		name: [
+	//		<name>: [
 	//			{
-	//				key: "anothervalue"
-	//				b: "xxx"
+	//				<key>: "anothervalue"
+	//				anotherKey: "xxx"
 	//			},
 	//			{
-	//				key: "value"  // This object is returned.
-	//				b: "yyy"
+	//				<key>: "<value>"
+	//				anotherKey: "yyy" // parentBoundValue gets a reference to this Json object
 	//			}
 	//		]
 	// }
 
 	for (const auto& parent : parentKeys)
 	{
 		found = false;
-		if (createAnyway && !result->isMember(parent.name))	(*result)[parent.name] = Json::Value(Json::arrayValue);
+		if (createAnyway && !parentBoundValue->isMember(parent.name))	(*parentBoundValue)[parent.name] = Json::Value(Json::arrayValue);
 
-		if (result->isMember(parent.name))
+		if (parentBoundValue->isMember(parent.name))
 		{
-			Json::Value& parentBoundValues = (*result)[parent.name];
+			Json::Value& parentBoundValues = (*parentBoundValue)[parent.name];
 			if (!parentBoundValues.isNull() && parentBoundValues.isArray())
 			{
 				for (Json::Value & boundValue : parentBoundValues)
```

Comment:
Should it not be `const Json::Value & boundValue`? 
Just to be sure you aren't changing other objects?